lib/pp.ml

```diff
@@ -297,3 +311,57 @@ let prvect_with_sep sep elem v = prvecti_with_sep sep (fun _ -> elem) v
 let prvect elem v = prvect_with_sep mt elem v
 
 let surround p = hov 1 (str"(" ++ p ++ str")")
+
+(*** DEBUG code ***)
+
+let db_string_of_pp pp =
+  let buf = Buffer.create 16 in
```

Comment:
a `create` of 16 here is fairly useless. I am not sure what to do about this printer, but certainly it should not use `buf` but fprintf.

Reply:
This function is used to print a Pp when an `assert_equal` fails in a unit test.  `assert_equal` requires a function that returns a string.  Are you suggesting that there be a db_print routine that goes directly to stdout?  I'm happy to create another routine that does that.

Indeed, the output string will generally be a lot longer than 16, but the length is a required parameter to create a buffer.  The doc for Buffer says "In doubt, take n = 16 for instance."  This isn't performance-critical code, either.

Your thoughts?

Reply:
I'd suggest you make this function to take a `fmt` and use `fprintf`. Then, the top level caller can chose the final destination.

As for the buffer, I'd use 256, relocating the buffer is expensive and wasteful. Anyways, if you switch to `fprintf` you won't have to care about that here.

Reply:
Ah, to get a string with the full output I have to use a buffer, right?

Printf.fprintf takes an out_channel.  fmt would be for Format.* routines, (I expect we don't want to call Format here.)  However, I'm not finding a routine to create an out_channel that writes to a buffer.  If you know of such, I'll use that, otherwise I'll have an inner routine that takes a curried value, either (Printf.fprintf outchan) or (Printf.bprintf buf) plus 2 wrapper functions to give a clean interface.

Reply:
Sorry I was not clear, I suggest you use `Format` indeed, it is the right tool for this kind of job.